### ISSUE TITLE
torch.exp2 backward overflows to inf at 1024.0 although the float64 derivative is finite

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#192407]
### 🐛 Describe the bug

`torch.autograd` returns `inf` for the gradient of `torch.exp2` at a `float64` input where the mathematically correct derivative is finite and representable in `float64`.

The derivative of `exp2(x)` is `ln(2) * 2^x`. For `x = 1024.0`, the true derivative is approximately `1.2460659279417838e+308`, which is below the maximum finite `float64` value. However, PyTorch returns `inf` for the gradient.

This issue reproduces on CPU, so CUDA availability does not affect the result.

```python
import torch

torch.set_default_dtype(torch.float64)

x = torch.tensor(1024.0, requires_grad=True)

y = torch.exp2(x)
y.backward()

actual = x.grad
expected = torch.tensor(1.2460659279417838e+308, dtype=torch.float64)

print("actual:")
print(actual)
print("expected:")
print(expected)
```

### Actual vs expected result

```text
actual:
tensor(inf)

expected:
tensor(1.2461e+308)
```

The discrepancy occurs in the backward pass. Although the forward value `torch.exp2(1024.0)` overflows, the mathematical derivative `ln(2) * 2^1024` is still finite in `float64`, so the gradient should not be `inf`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64

Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01

CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

**Comment by malfet** [comment]
Closing as duplicate of https://github.com/pytorch/pytorch/issues/192408
